Sources/MediaViewController.swift

```diff
@@ -112,20 +128,21 @@ public class VLCMediaViewController: UICollectionViewController, UISearchResults
                 backgroundview.clipsToBounds = true
             }
         }
-        if #available(iOS 11.0, *) {
-            navigationItem.searchController = searchController
-        } else {
-            navigationItem.titleView = searchController?.searchBar
-            searchController?.hidesNavigationBarDuringPresentation = false
-        }
+        showSearchController(true)
     }
-
+    
     func setupNavigationBar() {
         navigationItem.leftBarButtonItem = UIBarButtonItem(title: NSLocalizedString("SORT", comment: ""), style: .plain, target: self, action: #selector(sort))
     }
     
     func displayEmptyViewIfNeeded() {
-        collectionView?.backgroundView = collectionView?.numberOfItems(inSection: 0) == 0 ? emptyView : nil
+        if collectionView?.numberOfItems(inSection: 0) == 0 {
```

Comment:
hmm thinking about it we can probably move the searchbar code in here as well and adjust the methodname to reflect that it updates the UI for empty state or not

Reply:
yeah so just delete showSearchController  and change this Method to this : 
```
Bool isEmpty  =  collectionView?.numberOfItems(inSection: 0) == 0
collectionView?.backgroundView = isEmpty ? emptyView : nil
if #available(iOS 11.0, *) {
  navigationItem.searchController =  isEmpty ? nil : searchController
} else {
  navigationItem.titleView =  isEmpty ? nil : searchController?.searchBar
} 
```